src/accessories/characteristics/brightness.ts

```diff
@@ -62,8 +62,10 @@ export class BrightnessCharacteristic extends TuyaWebCharacteristic {
       data?.color?.brightness !== undefined
     ) {
       stateValue = Number(data.color.brightness);
-    } else if (data?.brightness) {
-      stateValue = Math.round((Number(data.brightness) / 255) * 100);
+    } else if (data?.brightness && data?.max_brightness) {
+      stateValue = Math.round(
+        (Number(data.brightness) / data.max_brightness) * 100
+      );
```

Comment:
`data?.max_brightness` will not be defined in case the user does not enter device defaults information.
Please use the old value (255) as the default value In this case.

```suggestion
    } else if (data?.brightness) {
      const max_brightness = data?.max_brightness ?? 255
      stateValue = Math.round(
        (Number(data.brightness) / max_brightness) * 100
      );
```

This will use 255 as fallback incase the `max_brightness` is not set.

Reply:
Actually I set the default value for max brightness to be 255 in the config file, then I read that value weather its provided or not, good enough? (check out latests commit)

Reply:
I believe you only set the "default" value in the ui config. This has nothing to do with the actual config file. Just when users use config-ui-x. Furthermore they can still remove the default value, making it undefined. Or not add the device config at all. Therefor I believe the proposed changes are still necessary.

Reply:
Done

Reply:
data?.max_brightness is null for me, before the change I read the value from the config file, where is the connection between data.max_brightness and config.max_brightness should be set?